core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1871,4 +1879,22 @@ void beforeExecution() {
     void afterExecution() {
         callCustomizers(c -> c.afterExecution(stmt, getContext()));
     }
+
+    private void attachJfrEvent(JdbiStatementEvent evt, StatementContext ctx) {
+        if (evt.shouldCommit()) {
+            evt.traceId = ctx.getTraceId();
+            evt.type = ctx.describeJdbiStatementType();
+            final var stmtConfig = getConfig(SqlStatements.class);
+            final String renderedSql = ctx.getRenderedSql();
+            if (renderedSql != null) {
+                evt.sql = renderedSql.substring(0,
+                        Math.min(renderedSql.length(), stmtConfig.getJfrSqlMaxLength()));
+            }
+            evt.parameters = getBinding().describe(stmtConfig.getJfrParamMaxLength());
+            ctx.addCleanable(() -> {
```

Comment:
this is surely a creative definition of "clean up" :-)

Reply:
It happens at the perfect time in the statement lifecycle, and is an already existing extension point. So it worked out really nicely in the end.